app/components/organisms/header/header.tsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import './header.scss';
+import Navbar from '../../molecules/navbar/navbar';
+import { usePathname } from 'next/navigation';
+
+interface currentPathInfo {
+  label: string;
+  href: string;
+};
+
+const Header: React.FC = () => {
+
+  const navItems = [
+    {label: 'HOME', href: '/'},
+    { label: 'LOGIN', href: '/login' },
+    { label: 'DASHBOARD', href: '/dashboard' },
+  ];
+
+  const pathname = usePathname() || '/';
+  const currentActivePage = navItems.find(item => (item.href === pathname));
```

Comment:
The reason for the type error is because the `find` method can return `undefined` if no values match: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find
You need to handle this by providing a default response.
```suggestion
  const currentActivePage = navItems.find(item => (item.href === pathname)) ?? your default;
```